_aggregations/pipeline/index.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: default
+title: Pipeline aggregations
+nav_order: 5
+has_children: true
+redirect_from:
+  - /opensearch/pipeline-agg/
+  - /query-dsl/aggregations/pipeline-agg/
+  - /query-dsl/aggregations/index/
+---
+
+# Pipeline aggregations
+
+Pipeline aggregations chain together aggregations by piping the result of one aggregation to the input of another. They compute complex statistical and mathematical measures like derivatives, moving averages, and cumulative sums. Some pipeline aggregations duplicate the functionality of metric and bucket aggregations, but in many cases are more intuitive to use.
+
+Pipeline aggregations are run after all other sibling aggregations. This has performance implications. For example, using the `bucket_selector` pipeline aggregation to narrow a list of buckets does not reduce computations performed on omitted buckets.
+{: .note}
+
+Pipeline aggregations cannot be sub-aggregated, but can be chained with other pipeline aggregations. For example, you calculate a second derivative by chaining two consecutive `derivative` aggregations. Keep in mind that pipeline aggregations append to existing output. So for example computing a second derivative by chaining `derivative` aggregations outputs both the first and second derivatives.
+
+## Types of pipeline aggregations
+
+Pipeline aggregations are of two types, sibling and parent.
+
+### Sibling aggregations
+
+A sibling pipeline aggregation takes the output of a nested aggregation and produces new buckets or new aggregations at the same level as the nested buckets.
+
+A sibling aggregation must be a multi-bucket aggregation (have multiple grouped values for a certain field) and the metric must be a numeric value.
+
+`min_bucket`, `max_bucket`, `sum_bucket`, and `avg_bucket` are common sibling aggregations.
+
+### Parent aggregations
+
+A parent aggregation takes the output of an outer aggregation and produces new buckets or new aggregations at the same level as the existing buckets.
+
+The specified metric for a parent aggregation must be a numeric value.
+
+We strongly recommend setting `min_doc_count` to `0` (the default for `histogram` aggregations) for parent aggregations. If `min_doc_count` is greater than `0` the aggregation omits buckets, which might lead to incorrect results.
+{: .important}
+
+`derivatives` and `cumulative_sum` are common parent aggregations.
+
+## Buckets path property
+
+A pipeline aggregation uses the `buckets_path` property to access the results of other aggregations.
+The `buckets_path` property has the following syntax:
+
+```
+buckets_path = <AGG_NAME>[<AGG_SEPARATOR>,<AGG_NAME>]*[<METRIC_SEPARATOR>, <METRIC>];
```

Comment:
I'm confused with the syntax here.

In later example-
For example, `my_sum.sum` selects the `sum` metric of an aggregation called `my_sum`. `popular_tags>my_sum.sum` nests `my_sum.sum` into the `popular_tags` aggregation.

I cannot understand how to construct the buckets_path.

Even in later example `"buckets_path": "number_of_bytes>_count"` I cannot understand the components (agg name, metric separator, metric) correctly with the syntax.

Reply:
I'm not happy with this explanation either. It looks like a sloppy combination of BNF and regex syntax. Even with the explanation key (below), it's hard to follow.

I'd like to rewrite this, but I'm not sure how yet. It might be a matter of actually putting it into BNF, or of writing a more complete key to the syntax elements.

It would be good if we were consistent across the documentation with this type of explanation. Do you know of anywhere else in the documentation where there's a good explanation of expression syntax? How is it done?